question16/main.js

```diff
@@ -0,0 +1,106 @@
+    return json.data;
+  }
+  catch(error) {
+    tabsGroup.textContent = "データが取得できませんでした";
+    console.error(error);
+  }
+  finally {
+    console.log('CONPLETE!');
+  }
+}
+
+async function init() {
+  const result = await callApi();
+  if (result){
+    renderElement(result);
+    } else {
+    tabsGroup.textContent = "データの表示に失敗しました";
+    console.error('データの表示に失敗しました'); 
+  }
+}
+init();
+
+//Rendering tabs and contents
+function renderElement(data){
+    newsWrapper.appendChild(contentsWrapper);
+    const length = data.length;
+    const fragmentTablists = document.createDocumentFragment();
+    const fragmentContents = document.createDocumentFragment();
+
+    //Display category names in tabs
+    for (let i = 0; i < length; i++){
+      const {category, articles, img} = data[i];
+      const tabLists = document.createElement('li');
+     
+      tabLists.innerHTML = category;
+      tabLists.classList.add('tabList');
+      fragmentTablists.appendChild(tabLists);
+      
+      const contentsContainer = createElementWithClassName("div", "contentsContainer");
+      const tabContents = createElementWithClassName("div", "tabContents");
+      const tabContentsUl = createElementWithClassName("ul", "tabContentsUl");
+
+      //Get the title of articles
+      for (let j = 0; j < articles.length; j++){
+        const articleTitle = data[i].articles[j].title;
+        const titleList = document.createElement('li');
+        const articleLink = document.createElement('a');
+        articleLink.href = "#";
+        articleLink.innerHTML = articleTitle;
+        fragmentContents.appendChild(contentsContainer).appendChild(tabContents).appendChild(tabContentsUl).appendChild(titleList).appendChild(articleLink);
+      }
+
+      //Display the img
+      const imgWrapper = createElementWithClassName("div", "imgWrapper");
+      const imgTag = document.createElement('img');
+      imgTag.src = img;
+      imgWrapper.appendChild(imgTag);
+      tabContents.appendChild(imgWrapper);
+
+      //Display the first content.
+      if (i === 0){
+        tabLists.classList.add('is-active');   
+        contentsContainer.classList.add('is-show');
+      }
+    }
+    tabsGroup.appendChild(fragmentTablists);
+    contentsWrapper.appendChild(fragmentContents);
+}
+
+
+//Tab switching function
+tabsGroup.addEventListener("click", (e) => {
+  const activeElement = document.getElementsByClassName("is-active")[0];
+  const showElement = document.getElementsByClassName('is-show')[0];
+
+  if (activeElement && showElement && e.currentTarget !== e.target) {
+    activeElement.classList.remove("is-active");
+    e.target.classList.add("is-active");
+
+      const tabList = document.getElementsByClassName('tabList');
+      const index = Array.prototype.indexOf.call(tabList,e.target);
+      showElement.classList.remove('is-show'); 
+      document.getElementsByClassName('contentsContainer')[index].classList.add('is-show'); 
+  }
+});
```

Comment:
Initially, I used this article as a reference to build the content.
https://blog-and-destroy.com/24218

To switch tabs, get the index number of the clicked "tabList" and add the "is-show" class to the "contentsContainer" with the same index number.

Since the HTML is already built in the article, I needed to use “e.target” this time to get the clicked element.

Did I answer your question..? 

> And when I tried deleting this if statement, it still worked.
➡️Really..!! I think I could not understand very well..! Thank you so much for pointing out!!I deleted it! 
Latest commit : https://github.com/csakyo/homework/commit/583f0fed25f0a743b7d25f48f6131dfe731f897a

Reply:
Thank you so much for this reference! Actually, I couldn’t understand about this if statement meaning but I didn't ask you for details😭  It's my fault, I'm sorry for the confusion💦 

This is just IMO.
Now you use `activeElement` and `showElement` as variable names, but I thought it's a little hard to tell which is which. So how about using a more specific name like `activeTab` etc? :)

Reply:
I see! No problem at all!
In fact, I've lost track of the article I was referring to...! Maybe I wanted the function to be executed when the inactive element was clicked, so I wrote it this way. (But I am not sure...) I'm sorry I don't understand  even though I wrote the code myself... please ignore:sweat: 

>Now you use activeElement and showElement as variable names, but I thought it's a little hard to tell which is which. So how about using a more specific name like activeTab etc? :)

Yes it is true!! 
I fixed the name from ```activeElement```and ```showElement``` to ``` activeTab``` and ```activeContent ```!
How about it?

Latest commit : https://github.com/csakyo/homework/pull/29/commits/dab82d6d1caec8f7f4a68459c55f0f4c91a96221